scripts/download-integration-test-binaries.sh

```diff
@@ -206,6 +206,7 @@ cleanup() {
 
 setup() {
 	cleanup
+	rm -rf ${BIN_DIR}
```

Comment:
What will it delete?

Reply:
The `BIN_DIR` is the final location where the binaries are stored. The reason I delete this folder before creating it is that if the folder is not empty, running make prepare_test_binaries will result in some strange tar command errors, as shown in the image below. This causes make prepare_test_binaries to fail unless BIN_DIR is manually deleted beforehand.

![CleanShot 2024-09-26 at 17 08 55@2x](https://github.com/user-attachments/assets/2cc08473-c131-429b-b66a-76186d543801)

Reply:
I think to avoid repeatly downloading and reduce waiting time, we should not delete the whole folder. It's better to ignore existing files one-by-one, and use `mv` to atomically move the files.

Reply:
This problem is not caused by the `mv` command but by the `tar` command. The error occurs when extracting `pd-server.tar.gz`. The reason is that the script needs to extract all the binary files under the `bin` directory from `pd-server.tar.gz`, so a wildcard `"bin/*"` is used to extract the contents of the `bin/*` directory. However, if there is already a `bin` folder in the current working directory and it contains files that do not exist in the `bin` directory of `pd-server.tar.gz`, the error will occur. This happens because before `tar` is executed, the wildcard is expanded, so it reads all the contents of the current working directory's `bin` folder and passes them as arguments to `tar`, which then tries to extract these files that are not present in the archive, leading to the error.

Additionally, I have also tried not using the wildcard and directly extracting the `bin` directory from `pd-server.tar.gz`. However, if the `BIN_DIR` already contains a folder with the same name, the `mv` command will fail:
![CleanShot 2024-09-29 at 11 57 07@2x](https://github.com/user-attachments/assets/b564f0c3-014d-4ece-9a85-a0ab55966d6e)

Therefore, the simplest and most effective solution is to directly clean the `BIN_DIR` before proceeding. This way, the script can be executed repeatedly without causing any issues. It will not have any negative impact since `BIN_DIR` does not exist by default and is `gitignored` in the project. It is created specifically by the script for this purpose.

Reply:
Oh, I see that `prepare_test_binaries` is manually executed. I wrongly think it will be executed automatically in integration test so I don't want the cached binaries to be deleted.

BTW, to achieve "atomic" extracting, we can extract them to a temporary folder in the same file system mounting entry and `mv` the folder.